datafusion/src/logical_plan/dfschema.rs

```diff
@@ -149,47 +147,80 @@ impl DFSchema {
         )))
     }
 
-    /// Find the index of the column with the given qualifer and name
-    pub fn index_of_column(&self, col: &Column) -> Result<usize> {
-        for i in 0..self.fields.len() {
-            let field = &self.fields[i];
-            if field.qualifier() == col.relation.as_ref() && field.name() == &col.name {
-                return Ok(i);
-            }
+    fn index_of_column_by_name(
```

Comment:
Refactored this code into its own helper method to reduce duplicated code between `field_with_qualified_name` and `index_of_column`. This should also make `index_of_column` more robust.